numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -621,6 +624,22 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self){
+    PyObject *arr, *ret = NULL;
+
+    if (!PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
+        return gentype_str(self);
```

Comment:
I think here the use of the same function for `repr` and `str` for void will fail:  wouldn't you want to call `gentype_repr` if this was called as `repr(void_scalar)`?